components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/model/ParRequestCacheEntry.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.par.model;
+
+import org.wso2.carbon.identity.core.cache.CacheEntry;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * PAR request with all attributes for caching.
+ */
+public class ParRequestCacheEntry extends CacheEntry {
+
+    private String requestUri;
+    private Map<String, String> params;
+    private long scheduledExpiryTime;
+    private String clientId;
+
+
+    /**
+     * Contractor for Cache entry of a PAR request.
+     *
+     * @param requestUri parameter map
+     * @param params scheduled expiry time
+     * @param scheduledExpiryTime client id
+     */
+    public ParRequestCacheEntry(String requestUri, Map<String, String> params, long scheduledExpiryTime) {
+
+        this.requestUri = requestUri;
+        this.params = params;
+        this.scheduledExpiryTime = scheduledExpiryTime;
+        this.clientId = params.get(OAuthConstants.OAuth20Params.CLIENT_ID);
```

Comment:
Lets pass in the client_id  to the method without extracting it from the params. That will help avoid any issues if the params doesn't contain a client_id param